devito/archinfo.py

```diff
@@ -60,16 +61,27 @@ def get_cpu_brand():
             except:
                 pass
 
-            return None
+            try:
+                # Certain ARM CPUs, e.g. Marvell Thunder X2
+                return cpuinfo.get_cpu_info().get('arch').lower()
+            except:
+                pass
 
         cpu_info['flags'] = get_cpu_flags()
         cpu_info['brand'] = get_cpu_brand()
-
-    if not all(i in cpu_info for i in ('flags', 'brand')):
-        # Fallback
-        ci = cpuinfo.get_cpu_info()
-        cpu_info['flags'] = ci.get('flags')
-        cpu_info['brand'] = ci.get('brand')
+    else:
```

Comment:
what happens if we get into the `if lines`  branch but either of `flags` or `brand` are None? 

are we sure that this new version behaves like the previous one? I'm not

Reply:
Added `return None` in the last try-except pair of each `get_cpu_xxx` function. I think now it has the same behaviour as before.